src/main/java/example/ExamplePlugin.java

```diff
@@ -20,6 +27,149 @@ public ExamplePlugin(){
                 Call.sendMessage("[scarlet]ALERT![] " + ((Player)event.builder).name + " has begun building a reactor at " + event.tile.x + ", " + event.tile.y);
             }
         });
+
+        //run when world loaded
+        Events.on(WorldLoadEvent.class, event -> {
+            //Log.info("Game over!");
+        });
+
+        //run when player connected
+        Events.on(PlayerConnect.class, event -> {
+            //Log.info(event.player.name+" player connected.");
+        });
+
+        //If the player has moved the resource directly
+        Events.on(DepositEvent.class, event -> {
+            // If deposit block name is thorium reactor
+            if(event.tile.block() == Blocks.thoriumReactor){
+                // Prevent the main thread from hanging when thread.sleep
+                Thread t = new Thread(() -> {
```

Comment:
Absolutely never do this. Doing anything in a new thread will cause synchronization issues and crashes. This exact code was most likely what caused that crash you reported in #bugs. If you just want to run something with a delay, do `Time.runTask(...)` or `Time.run(...)`